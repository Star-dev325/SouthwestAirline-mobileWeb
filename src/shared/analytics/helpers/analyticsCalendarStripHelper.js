// @flow
import dayjs from 'dayjs';

import type { BoundType } from 'src/shared/flow-typed/flightBound.types';
import { INBOUND, OUTBOUND } from 'src/shared/constants/flightBoundTypes';

export const calculateDateDiffs = (
  newDate: string,
  previousDate: string,
  direction: BoundType,
  currentDiffs: { [BoundType]: string } = {}
): { [BoundType]: string } => {
  const diff = dayjs(newDate).diff(previousDate, 'days');
  const sign = Math.sign(diff) === 1 ? '+' : '';
  const diffs = { ...currentDiffs };

  diffs[direction] = `${sign}${diff}`;

  return diffs;
};

export const calculateOverrideDateDiffs = (
  newDate: string,
  previousDate: string,
  direction: BoundType,
  isOverrideEndDate: boolean,
  previousDiffs: { [BoundType]: string } = {}
): { [BoundType]: string } => {
  let diffs = {};

  if (direction === OUTBOUND) {
    if (isOverrideEndDate) {
      diffs = calculateDateDiffs(newDate, previousDate, INBOUND, previousDiffs);
    } else {
      diffs = {};
    }
  }

  return diffs;
};
