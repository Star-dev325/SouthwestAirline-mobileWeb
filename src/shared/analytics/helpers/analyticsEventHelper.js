import _ from 'lodash';

const pageIdentifierTransformer = [
  { matcher: /^\/$|^\/\?/, id: 'home' },
  { matcher: /\/flight-status\/(?!.*html).*/, id: 'flight-status-results' },

  { matcher: /\/flight-status\?.*_modal=originAirport/, id: 'flight-status-from' },
  { matcher: /\/flight-status\?.*_modal=destinationAirport/, id: 'flight-status-to' },

  { matcher: /\/air\/booking\/shopping\?.*_modal=from/, id: 'air-booking-shopping-from' },
  { matcher: /\/air\/booking\/shopping\?.*_modal=to/, id: 'air-booking-shopping-to' },
  { matcher: /\/air\/booking\/shopping\?.*_modal=airBookingCalendar/, id: 'air-booking-shopping-dates' },

  { matcher: /\/air\/change\/shopping\?.*_modal=from/, id: 'air-change-shopping-from' },
  { matcher: /\/air\/change\/shopping\?.*_modal=to/, id: 'air-change-shopping-to' },
  { matcher: /\/air\/change\/shopping\?.*_modal=airBookingCalendar/, id: 'air-change-shopping-dates' },

  { matcher: /\/car\/booking\?.*_modal=pickUpModal/, id: 'car-booking-pick-up' },
  { matcher: /\/car\/booking\?.*_modal=dropOffModal/, id: 'car-booking-drop-off' },
  { matcher: /\/car\/booking\?.*_modal=carBookingCalendar/, id: 'car-booking-dates' },

  { matcher: /\/check-in\/reservation\/\w{6}$/, id: 'checkin-view-reservation' }
];

const _matchPath = (path) =>
  _.get(
    _.find(pageIdentifierTransformer, (t) => path.match(t.matcher)),
    'id'
  );

export const transformPath = (path, search = null) => {
  if (!path) {
    return '';
  }
  let toMatchPath = path;

  if (search) {
    toMatchPath = `${path}${search}`;
  }

  return _matchPath(toMatchPath) || path.replace(/\//g, '-').slice(1);
};

export const raiseEvent = (eventName) => {
  window._trackAnalytics && window._trackAnalytics.event();

  const customEvent = new window.CustomEvent(eventName);

  document.body.dispatchEvent(customEvent);
};

export const raiseSatelliteEvent = (eventName, detail) => {
  if (window._satellite) {
    detail ? window._satellite.track(eventName, detail) : window._satellite.track(eventName);
  }
};

export const customerMessageAnalytics = (messages) => {
  if (messages) {
    const messageKeys = _.chain(messages).map('key').compact().join().value();

    _.set(window, 'data_a.message.customer', messageKeys);
    _.set(window, 'data_a.message.customerdisplay', '1');
  }
};

export const getPageIdentifier = (location) => {
  const { pathname = '', search = '' } = location || {};

  return transformPath(pathname, search);
};
