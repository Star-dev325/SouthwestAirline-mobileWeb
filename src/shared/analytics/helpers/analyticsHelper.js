import _ from 'lodash';

export const generateUpdatedFlowStoreForAnalytics = (flowSelectors, state, actionType) =>
  _.chain(flowSelectors)
    .pickBy((selectorObj) => _.includes(selectorObj.actions, actionType))
    .mapValues((selectorObj) => selectorObj.selector(state))
    .value();

export const generateFlowActionListForAnalytics = (flowSelectors) =>
  _.chain(flowSelectors).mapValues('actions').values().flatten().uniq().value();

export const generateUpdatedStoresForAnalytics = (generators, state, actionType) =>
  _.reduce(
    generators,
    (result, generator, key) => {
      _.each(generator(state, actionType), (subValue, subKey) => {
        result[`${key}.${subKey}`] = subValue;
      });

      return result;
    },
    {}
  );
