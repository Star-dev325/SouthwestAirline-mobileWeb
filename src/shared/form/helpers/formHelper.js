// @flow
const repeatStr = (str, times = 1) => {
  let result = str;

  while (times > 1) {
    result += str;
    times--;
  }

  return result;
};

export const getMaskProps = function (mask: string | { rule: string, repeat?: number, maskChar?: string }) {
  const defaultMask = {
    repeat: 1,
    maskChar: null
  };

  const maskConfigs = Object.assign({}, defaultMask, typeof mask === 'string' ? { rule: mask } : mask);

  const finalMaskStr = repeatStr(maskConfigs.rule, maskConfigs.repeat);

  return {
    mask: finalMaskStr,
    maskChar: maskConfigs.maskChar,
    maxLength: finalMaskStr.length // mask can't work properly in some android device so use maxLength to limit the user input, see bug#2026
  };
};
