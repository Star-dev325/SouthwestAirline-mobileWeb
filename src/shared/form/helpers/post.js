import _ from 'lodash';
import BrowserObject from 'src/shared/helpers/browserObject';

const { document } = BrowserObject;

const post = (formProps, params) => {
  const createForm = () => {
    const form = document.createElement('form');
    const { target } = formProps;

    form.setAttribute('method', formProps.method || 'POST');
    form.setAttribute('action', formProps.path);

    if (target) {
      form.setAttribute('target', target);
    }

    return form;
  };

  const createSubmitButton = () => {
    const submitButton = document.createElement('input');

    submitButton.setAttribute('type', 'submit');
    submitButton.setAttribute('style', 'display: none;');
    submitButton.setAttribute('value', 'submitButton');

    return submitButton;
  };

  const createInput = (key, value) => {
    const input = document.createElement('input');

    input.setAttribute('type', 'hidden');
    input.setAttribute('name', key);
    input.setAttribute('value', value);

    return input;
  };

  if (formProps && formProps.path) {
    const form = createForm();

    _.each(params, (value, key) => {
      form.appendChild(createInput(key, params[key]));
    });

    form.appendChild(createSubmitButton());
    document.body.appendChild(form);
    form.submit();
    form.parentNode.removeChild(form);
  }
};

export default post;
