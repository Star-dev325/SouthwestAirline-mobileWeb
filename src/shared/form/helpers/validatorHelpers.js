// @flow
import _ from 'lodash';
import validator from 'src/shared/form/formValidators/validator';
import {
  REQUIRED_ERROR,
  DEFAULT_ERROR_TYPE,
  FIELD_ERROR_MESSAGE
} from 'src/shared/form/constants/validationErrorTypes';
import type {
  FormData,
  ValidationRule,
  FormValidationRules,
  FieldValidationRules,
  FormValidationErrors
} from 'src/shared/form/flow-typed/form.types';

function executeRules(rules, ...args) {
  return _.reduce(
    rules,
    (result, { validator: ruleValidator, type = DEFAULT_ERROR_TYPE, msg }) => {
      if (_.isNull(result)) {
        const isValid = ruleValidator && ruleValidator(...args);

        if (_.isPromise(isValid)) {
          const promise = isValid;

          return { promise, type, msg };
        }

        return isValid ? null : { type, msg };
      }

      return result;
    },
    null
  );
}

export function getErrorMsgForField(fieldName: string, fieldValue: *, fieldRules: ValidationRule[]) {
  const { true: requiredRules, false: otherRules } = _.groupBy(fieldRules, (rule) => !!rule.isRequired);
  const isRequiredField = !_.isEmpty(requiredRules);

  if (isRequiredField && !validator.isRequired(fieldValue)) {
    const { msg } = requiredRules[0];

    if (msg) {
      return {
        msg,
        type: FIELD_ERROR_MESSAGE
      };
    } else {
      return {
        type: REQUIRED_ERROR
      };
    }
  }

  if (!isRequiredField && _.isEmpty(fieldValue)) {
    return null;
  }

  return executeRules(otherRules, fieldValue);
}

function hasAnyAsycValidation(errors) {
  return _.some(errors, (error) => _.isPromise(error.promise));
}

function waitAsyncValidationResolvedAndMergeResult(errors) {
  const errorsWithPromise = _.chain(errors)
    .pickBy((error) => _.isPromise(error.promise))
    .entries()
    .value();
  const asyncErrorKeys = _.map(errorsWithPromise, '[0]');
  const asyncErrorContents = _.map(errorsWithPromise, '[1]');

  return Promise.all(_.map(asyncErrorContents, 'promise')).then((asyncErrorValues) => {
    const asyncErrorContentWithValidationResult = _.map(asyncErrorValues, (asyncErrorValue, index: number) => {
      if (!_.isEmpty(asyncErrorValue)) {
        const { type } = asyncErrorContents[index];

        return { type, msg: asyncErrorValue };
      }

      return null;
    });
    const asyncErrors = _.zipObject(asyncErrorKeys, asyncErrorContentWithValidationResult);

    return _.chain(errors).assign(asyncErrors).pickBy(_.negate(_.isEmpty)).value();
  });
}

export function executeValidators(
  formData: FormData,
  formRules: FormValidationRules = {},
  fieldRules: FieldValidationRules = {}
): FormValidationErrors | Promise<FormValidationErrors> {
  const fieldErrors = _.chain(formData)
    .mapValues((fieldValue, fieldName) => getErrorMsgForField(fieldName, fieldValue, fieldRules[fieldName]))
    .pickBy(_.negate(_.isNull))
    .value();

  let errors = fieldErrors;

  if (!_.isEmpty(formRules)) {
    const formErrors = _.chain(formRules)
      .mapValues((rules) => executeRules(rules, formData, fieldErrors))
      .pickBy(_.negate(_.isNull))
      .value();

    errors = _.merge(errors, formErrors);
  }

  if (!hasAnyAsycValidation(errors)) {
    return errors;
  }

  return waitAsyncValidationResolvedAndMergeResult(errors);
}
