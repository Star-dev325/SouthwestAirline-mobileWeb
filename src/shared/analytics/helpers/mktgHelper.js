// @flow
import _ from 'lodash';
import mktgDataConstants from 'src/shared/constants/mktgDataConstants';
import { createSelector } from 'src/shared/helpers/createSelectorWithErrorReporter';
import { getGlobalMktgData } from 'src/shared/analytics/selectors/getGlobalMktgData';
import { toNumberStringWithoutCommas } from 'src/shared/helpers/currencyValueHelper';
import type { FlightProductCard, ProductDefinition } from 'src/shared/flow-typed/shared.types';
import type { MKTGFareType } from 'src/shared/flow-typed/mktg.types';

const {
  NOT_AVAILABLE,
  NOT_SELECTED,
  DIRECT_FLIGHT,
  POINTS_BOOKING,
  CURRENCY_BOOKING,
  CONNECTING_FLIGHT,
  NONSTOP_FLIGHT
} = mktgDataConstants;

export const parseFareValue = (fare: MKTGFareType, fareValue: (fare: MKTGFareType) => string) =>
  (typeof fare === 'string' && fare) || (fare.reasonIfUnavailable && NOT_AVAILABLE) || fareValue(fare);

export const getFarePrice = (fare: MKTGFareType) => {
  const pricePath = fare.discountedPrice ? 'discountedPrice.amount' : 'price.amount';
  const farePrice = _.get(fare, pricePath, NOT_AVAILABLE);

  return (
    (farePrice === NOT_AVAILABLE && farePrice) ||
    (parseInt(farePrice, 10) === 0 && 'zero') ||
    toNumberStringWithoutCommas(farePrice)
  );
};

export const getFarePriceDifference = ({ priceDifference: { amount = NOT_AVAILABLE, sign = '' } = {} }: MKTGFareType) =>
  (amount === NOT_AVAILABLE && amount) ||
  (parseInt(amount) === 0 && 'zero') ||
  (sign === '-' && `${sign}${toNumberStringWithoutCommas(amount)}`) ||
  toNumberStringWithoutCommas(amount);

export const getSortedFareProducts = (products: Array<ProductDefinition>): Array<ProductDefinition> =>
  [...products].sort((product1, product2) => (product2.rowOrder > product1.rowOrder ? 1 : -1));

export const getSortedFares = (fares: Array<MKTGFareType>, products: Array<ProductDefinition>): Array<MKTGFareType> =>
  products.map(
    ({ productId }) =>
      fares.find((fare) => typeof fare !== 'string' && fare._meta.fareProductId === productId) || NOT_AVAILABLE
  );

export const getFareData = (
  propPrefix: string,
  fares: Array<MKTGFareType>,
  products: Array<ProductDefinition>,
  bookingType: string,
  fareValue: (MKTGFareType) => string = getFarePrice
) =>
  products.reduce((fareData, product, index) => {
    const fare = fares.find(
      (fareItem) => typeof fareItem !== 'string' && fareItem._meta.fareProductId === product.productId
    );
    const {
      [`${propPrefix}_viewfaretypes`]: fareTypes,
      [`${propPrefix}_viewfareproductids`]: fareIds,
      [`${propPrefix}_viewfarecurrency`]: fareCurrencyList,
      [`${propPrefix}_viewfarepoints`]: farePointsList
    } = fareData;
    const fareValueResult =
      (fare && parseFareValue(fare, fareValue)) ||
      (fares.every((fareItem) => fareItem === NOT_SELECTED) && NOT_SELECTED) ||
      NOT_AVAILABLE;
    const fareCurrency = bookingType === CURRENCY_BOOKING ? fareValueResult : NOT_SELECTED;
    const farePoints = bookingType === POINTS_BOOKING ? fareValueResult : NOT_SELECTED;
    const { id: fareType = NOT_AVAILABLE, productId: fareProductId = NOT_AVAILABLE } = product;
    const dataIndex = index + 1;

    return {
      ...fareData,
      [`${propPrefix}_viewfaretypes`]: fareTypes ? `${fareTypes}|${fareType}` : fareType,
      [`${propPrefix}_viewfareproductids`]: fareIds ? `${fareIds}|${fareProductId}` : fareProductId,
      [`${propPrefix}_viewfarecurrency`]: fareCurrencyList ? `${fareCurrencyList}|${fareCurrency}` : fareCurrency,
      [`${propPrefix}_viewfarepoints`]: farePointsList ? `${farePointsList}|${farePoints}` : farePoints,
      [`${propPrefix}_viewfareproductid${dataIndex}`]: fareType,
      [`${propPrefix}_viewfaretype${dataIndex}`]: fareType,
      [`${propPrefix}_viewfarecurrency${dataIndex}`]: fareCurrency,
      [`${propPrefix}_viewfarepoints${dataIndex}`]: farePoints
    };
  }, {});

export const getFareDifferencesData = (propPrefix1: string, propPrefix2: string, fareDifferences: Array<string>) =>
  fareDifferences.reduce(
    (result, difference, index) => ({
      ...result,
      [`${propPrefix1}_viewfarediff${propPrefix2}${index + 1}`]: difference
    }),
    {}
  );

export const getStopData = (propPrefix: string, flight: FlightProductCard) => {
  const numberOfStops = _.get(flight, '_meta.numberOfStops', NOT_SELECTED);
  const hasStop = parseInt(numberOfStops) > 0;
  const hasPlaneChange = !!flight.stopCity;

  const stopType = hasPlaneChange ? CONNECTING_FLIGHT : hasStop ? DIRECT_FLIGHT : NONSTOP_FLIGHT;

  return {
    [`${propPrefix}_stops`]: numberOfStops,
    [`${propPrefix}_stoptype`]: numberOfStops === NOT_SELECTED ? NOT_SELECTED : stopType
  };
};

export const getMktgDataForPage = (path: string, defaultValue: *) => (state: *) => _.get(state, path, defaultValue || {});

export const createMktgDataSelector = (path: string, defaultValue: *) => createSelector(
  [getGlobalMktgData, getMktgDataForPage(path, defaultValue)],
  (globalMktgData, pageMktgData) => ({ ...globalMktgData, ...pageMktgData })
);
