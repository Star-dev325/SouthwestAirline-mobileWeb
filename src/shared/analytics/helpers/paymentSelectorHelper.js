import _ from 'lodash';

import { getCardType } from 'src/shared/helpers/creditCardHelper';

import {
  NEW_CREDIT_CARD_ID,
  RAPID_REWARDS_VISA_ID,
  PAY_PAL_CARD_ID,
  APPLE_PAY_CARD_ID,
  UPLIFT_CARD_ID
} from 'src/shared/constants/creditCardConstants';
import { INSTANT_CREDIT_RAPID_REWARDS_VISA, PAYPAL, APPLE_PAY, UATP } from 'src/shared/constants/creditCardTypes';
import { addCurrency } from 'src/shared/api/helpers/currencyHelper';
import { isCurrencyAmountZero } from 'src/shared/helpers/travelFundsHelper';

export const buildPaymentAnalyticsData = (
  paymentInfo,
  savedCreditCards,
  flightConfirmationPage,
  travelFundsBalanceRemaining
) => {
  let typeOfCreditCard;
  const { city, stateProvinceRegion, zipOrPostalCode, saveCreditCard, cardNumber } = paymentInfo;
  const fundsApplied = _.get(flightConfirmationPage, 'fundsApplied');
  let selectedCardId = _.get(paymentInfo, 'selectedCardId');
  const selectedSavedCreditCard =
    _.find(_.chain(savedCreditCards).values().flatten().value(), { savedCreditCardId: selectedCardId }) || {};
  const creditCardType = getCardType(cardNumber);
  const primaryCardType = _.get(savedCreditCards, 'primaryCard.type');
  const rtfs = [];
  const luvVouchers = [];
  const giftCards = [];
  const isApplePay = _.isEqual(selectedCardId, APPLE_PAY_CARD_ID);
  const isUplift = _.isEqual(selectedCardId, UPLIFT_CARD_ID);

  if (!_.isEmpty(fundsApplied)) {
    let fundTypes = [];

    if (!_.isEmpty(paymentInfo) && !isCurrencyAmountZero(travelFundsBalanceRemaining)) {
      if (_.isEmpty(creditCardType) && selectedSavedCreditCard.type) {
        fundTypes = [selectedSavedCreditCard.type];
      } else if (isApplePay) {
        fundTypes = [APPLE_PAY.shortName];
      } else if (isUplift) {
        fundTypes = [UATP.shortName];
      } else if (_.isEmpty(creditCardType)) {
        fundTypes = [INSTANT_CREDIT_RAPID_REWARDS_VISA.key];
      } else {
        fundTypes = [creditCardType];
      }
    } else if (_.isEmpty(paymentInfo) && !isCurrencyAmountZero(travelFundsBalanceRemaining)) {
      fundTypes = [primaryCardType];
    }

    _.forEach(fundsApplied, (fund) => {
      const fundType = _.get(fund, 'travelFundType');
      const appliedAmount = _.get(fund, 'appliedAmount');

      !_.includes(fundTypes, fundType) && fundTypes.push(fundType);
      fundType === 'TRAVEL_FUNDS' && rtfs.push(appliedAmount);
      fundType === 'LUV_VOUCHER' && luvVouchers.push(appliedAmount);
      fundType === 'GIFT_CARD' && giftCards.push(appliedAmount);

      typeOfCreditCard = fundTypes.join(',');
    });
  } else {
    if (_.isEmpty(paymentInfo)) {
      typeOfCreditCard = isApplePay ? APPLE_PAY.shortName : primaryCardType;
      typeOfCreditCard = isUplift ? UATP.shortName : typeOfCreditCard;
      selectedCardId = _.get(savedCreditCards, 'primaryCard.savedCreditCardId');
    } else {
      typeOfCreditCard = _.isEmpty(creditCardType)
        ? selectedSavedCreditCard.type || INSTANT_CREDIT_RAPID_REWARDS_VISA.key
        : creditCardType;

      typeOfCreditCard = isApplePay ? APPLE_PAY.shortName : typeOfCreditCard;
      typeOfCreditCard = isUplift ? UATP.shortName : typeOfCreditCard;
    }
  }

  const type = selectedCardId === PAY_PAL_CARD_ID ? PAYPAL.key : typeOfCreditCard;

  return _.omitBy(
    {
      city,
      state: stateProvinceRegion,
      zipcode: zipOrPostalCode,
      saveNewCardSelected: _.toBoolean(saveCreditCard) || false,
      storedCard:
        selectedCardId &&
        !isCurrencyAmountZero(travelFundsBalanceRemaining) &&
        !_.includes(
          [NEW_CREDIT_CARD_ID, RAPID_REWARDS_VISA_ID, PAY_PAL_CARD_ID, APPLE_PAY_CARD_ID, UPLIFT_CARD_ID],
          selectedCardId
        ),
      type,
      travelfund:
        rtfs.length > 1
          ? _.get(addCurrency(...rtfs), 'amount')
          : !_.isEmpty(rtfs[0])
            ? _.get(rtfs[0], 'amount')
            : undefined,
      luvvoucher:
        luvVouchers.length > 1
          ? _.get(addCurrency(...luvVouchers), 'amount')
          : !_.isEmpty(luvVouchers[0])
            ? _.get(luvVouchers[0], 'amount')
            : undefined,
      giftcard:
        giftCards.length > 1
          ? _.get(addCurrency(...giftCards), 'amount')
          : !_.isEmpty(giftCards[0])
            ? _.get(giftCards[0], 'amount')
            : undefined
    },
    _.isUndefined
  );
};
