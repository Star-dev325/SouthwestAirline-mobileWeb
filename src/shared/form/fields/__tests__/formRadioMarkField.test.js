import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import FormRadioMarkField from 'src/shared/form/fields/formRadioMarkField.jsx';
import _ from 'lodash';
import { createMockedForm, createMockedFormStore } from 'test/unit/helpers/formTestUtils';

describe('formRadioMarkField', () => {
  it('Should render radio field', () => {
    const  { container } = createComponent();

    fireEvent.click(container.querySelector('.flex-main-center'));
    expect(container).toMatchSnapshot();
  });
});

const createComponent = (props = {}) => {
  const defaultProps = {
    className: 'radio-1',
    callback: _.noop
  };

  const mergedProps = { ...defaultProps, ...props };

  const formStore = createMockedFormStore();
  const MockedForm = createMockedForm(formStore, props);

  return render(
    <MockedForm formData={{ firstbound: true }}>
      <FormRadioMarkField {...mergedProps} />
    </MockedForm>
  );
};
