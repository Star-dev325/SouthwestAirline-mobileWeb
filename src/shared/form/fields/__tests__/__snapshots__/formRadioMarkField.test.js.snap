// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formRadioMarkField Should render radio field 1`] = `
<div>
  <form>
    <div
      class="field"
    >
      <div
        class="radio-1"
      >
        <div
          class="flex2 flex flex-main-center"
        >
          <span
            class="radio-button--mark"
          >
            <span
              class="radio-input--mark active"
            />
          </span>
        </div>
      </div>
    </div>
  </form>
</div>
`;
